src/tools/openpgp-tool.c

```diff
@@ -52,22 +53,34 @@
 #define OPT_PIN     259
 #define OPT_DELKEY  260
 
+enum code_types {
+	TYPE_NULL,
+	TYPE_HEX,
+	TYPE_STRING
+};
+
 /* define structures */
 struct ef_name_map {
 	const char *name;
 	const char *env_name;
 	const char *ef;
-	char *(*prettify_value)(char *);
+	enum code_types type;
+	size_t offset;
+	size_t length;	/* 0 <=> potentially infinite */
+	char *(*prettify_value)(void *);
 };
 
 /* declare functions */
 static void show_version(void);
-static char *prettify_name(char *str);
-static char *prettify_language(char *str);
-static char *prettify_gender(char *str);
-static void display_data(const struct ef_name_map *mapping, char *value);
+static char *prettify_version(void *ptr);
+static char *prettify_manufacturer(void *ptr);
+static char *prettify_serialnumber(void *ptr);
+static char *prettify_name(void *ptr);
+static char *prettify_language(void *ptr);
+static char *prettify_gender(void *ptr);
```

Comment:
I don't understand why you're removing explicit typing and use `void *ptr` instead. I don't see a real advantage and I fear this could lead to more problems.